src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const array = [];
+  let newObj = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newObj, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const type of action.keysToRemove) {
+          if (newObj[type]) {
+            delete newObj[type];
+          }
+        }
+        break;
+
+      case 'clear':
+        newObj = {};
+        break;
+
+      default:
+        newObj = {};
```

Comment:
You should handle the error somehow here, not just copy clear case. For example, throw an error letting user know that passed action's type is not supported
`throw new Error("This action type is not supported")`